src/api/meal_plan.py

```diff
@@ -1,89 +1,227 @@
-                "start_date": plan.start_date,
-                "end_date": plan.end_date,
-                "daily_calorie_goal": plan.daily_calorie_goal
-            }
-        )
-        meal_plan_id = result.scalar()
-    return {"meal_plan_id": meal_plan_id}
+    try:
+        with db.engine.begin() as connection:
+            validate_account = connection.execute(text(
+                    "SELECT id FROM accounts WHERE id = :account_id"),
+                    {"account_id": plan.account_id }
+                ).one_or_none()
 
+            if validate_account is None:
+                raise Exception("Account not found.")
 
-@router.post("/meal-plans/{meal_plan_id}/log")
-def log_meal(meal_plan_id: int, meal: MealLog):
-    with db.engine.begin() as connection:
-        connection.execute(
-            text("""
-                INSERT INTO meal_logs (meal_plan_id, date, recipe_id, quantity, meal_type)
-                VALUES (:meal_plan_id, :date, :recipe_id, :quantity, :meal_type)
-            """),
-            {
-                "meal_plan_id": meal_plan_id,
-                "date": meal.date,
-                "recipe_id": meal.recipe_id,
-                "quantity": meal.quantity,
-                "meal_type": meal.meal_type
-            }
+            result = connection.execute(
+                text("""
+                    INSERT INTO meal_plans (author_id, name, description)
+                    VALUES (:account_id, :name, :description)
+                    RETURNING id
+                """),
+                {
+                    "account_id": plan.account_id,
+                    "name": plan.name,
+                    "description": plan.description
+                }
+            )
+
+            meal_plan_id = result.scalar()
+            return {"meal_plan_id": meal_plan_id}
+    
+    except Exception as e:
+        raise HTTPException(status_code=404, detail=f"{e}")
+
+@router.delete("/{meal_plan_id}")
+def delete_meal_plan(meal_plan_id: int):
+    try:
+        with db.engine.begin() as connection:
+            validate = connection.execute(
+                text("""SELECT 1 FROM meal_plans WHERE id = :meal_plan_id"""),
+                {"meal_plan_id": meal_plan_id}
+            ).one_or_none()
+
+            if validate is None:
+                print(f"Meal Plan with ID {meal_plan_id} not found.")
+                raise Exception("Meal Plan not found.")
+
+            connection.execute(
+                text("""DELETE FROM plans_recipes WHERE meal_plan_id = :meal_plan_id"""),
+                {"meal_plan_id": meal_plan_id}
+            )
+            connection.execute(
+                text("""DELETE FROM meal_plans WHERE id = :meal_plan_id"""),
+                {"meal_plan_id": meal_plan_id}
+            )
+
+            return Response(status_code=200)
+    
+    except Exception as e:
+        raise HTTPException(
+            status_code=404, detail=f"An error occurred while deleting the meal plan: {e}"
```

Comment:
I'm not sure if 404 is the correct fallback. I think 400 would be more accurate as 404 represents something not being found, while 400 is a generic "something bad happened". Thoughts?

Reply:
(Also this applies to all try/catch's with a 404 as the fallback response)

Reply:
I think we should sit down and try and figure out the best way to represent all the exceptions and how to correctly throw them. We have a week so I can do some research on it.